### ISSUE TITLE
Unexpected Report PDU error when user does not have access to the requested security level

### ISSUE THREAD

**Issue by gavinaiken** [markabrahams/node-net-snmp#77]
Attempting to request a get with the command line snmp tools, when the user is not configured correctly for the requested security level on the agent, returns this error:

```
$ snmpget -v 3 -u USER -a md5 -A ****** -x aes -X ****** -l authPriv 172.27.240.15 1.3.6.1.2.1.1.1.0
snmpget: Unsupported security level
```

However the node-net-snmp library gives a much less helpful `Unexpected Report PDU`

**Comment by gavinaiken** [comment]
This seems to be documented in the RFC: https://github.com/markabrahams/node-net-snmp/blob/master/ref/rfc/v3/rfc3414.txt#L1490

I will try to get a PR ready to improve the error reporting for this case and possibly a few others.

**Comment by markabrahams** [comment]
Yes - good point - the RFC does have advice worth taking there!

Merged #78 into master and released in v2.5.10 of the npm.
